opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1000,21 +1001,32 @@ namespace Opm {
                 fip_.fip[i].resize(nc,0.0);
             }
 
-            for (int c = 0; c < nc; ++c) {
-                const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
+            ElementContext elemCtx(ebosSimulator_);
+            auto elemIt = elemCtx.gridView().template begin</*codim=*/0>();
+            const auto& elemEndIt = elemCtx.gridView().template end</*codim=*/0>();
+            for (; elemIt != elemEndIt; ++elemIt) {
```

Comment:
It is very well possible that this has been working as advertised for a while. (if yes, does it work properly with `Dune::CpGrid`?) In general, my impression is that the parallel stuff is pretty much an afterthought for most Dune people and it thus does not get tested as well as the rest. I cannot really blame them for this ;)

> Could you give it a shot?

yes, I'll try. a problem may be that I cannot really test this in parallel because of the other MPI issues we have...